README.md

```diff
@@ -240,11 +240,20 @@ If you would like to debug local or even remote deployment, follow the procedure
 
 <details><summary>Kubernetes debugger setup</summary>
 
- 1. Create debug build locally by exporting `DEBUG_BUILD=yes`:
-    ```bash
-    $ DEBUG_BUILD=yes make image
-    ```
-    Alternatively, debug build will also be created when the branch name contains `-debug` substring. This works locally with `make image` and in CircleCI.
+ 1. Create a debug build
+    * Create debug build locally by exporting `DEBUG_BUILD=yes`:
+        ```bash
+        $ DEBUG_BUILD=yes make image
+        ```
+    * To create a debug build for remote debugging,
+      * Open Makefile under /stackrox and find the line containing `DEBUG_BUILD ?= no`.
+      * Comment out that line and add `DEBUG_BUILD ?= yes` underneath it.
```

Comment:
Hm. The new instructions don't make much sense to me. All what I ever had to do was running the command provided above `DEBUG_BUILD=yes make image`.
![image](https://user-images.githubusercontent.com/537715/169295271-4aff1ef7-de5f-44f4-be0d-536251fb21ac.png)

Was it not producing debug image for you for some reason @charmik-redhat?
Try running it (`DEBUG_BUILD=yes make image`) and then see if the following two commands would output anything
```
docker run --rm -it --entrypoint=/bin/bash stackrox/main:$(make tag)
ls -lh /go/bin/dlv
```

Reply:
@msugakov `DEBUG_BUILD=yes make image` works when I am using `deploy-local.sh` to locally deploy after that. However, form what I know, building and deploying locally doesn't work well for those with M1 processor. The best way to debug currently if one cannot deploy locally is to add log statements and deploy on GKE provisioned from infra.

Making this change makes it possible to have a debug build deployed on GKE cluster using the `deploy.sh` script. Also, deploying on GKE with `deploy.sh` needs the image to be available on a quay.io registry which is done by circleCI build and as far as I know, devs do not have permission to manually push images to the registry.

I agree that adding steps to temporarily update Makefile and then resetting it before merge is risky and that should be removed. It is anyways possible for circleCI build to automatically set the DEBUG_BUILD env var if the branch name contains `-debug` substring and keeping just that in README should suffice.

Reply:
I was wondering how the build and deployment process works for folks using M1 Macs - I guess it doesn't work well at the moment.
How about instead of changing the CI and enabling remote debug deployments we focus on the root cause:
Fixing the build process and the deploy scripts to run on ARM64 / M1.
I know for a fact that there is general interest from the community as well (see https://github.com/stackrox/stackrox/issues/1614 as well as this [Slack Thread in the CNCF Slack](https://cloud-native.slack.com/archives/C01TDE3GK0E/p1652325070937549)). 
I suspect that new associates that join our team will be equipped with M1 Macs, so this problem will only grow in size with each new team member. 
cc-ing @jschnath for visibility in here, as he has been poking around in our CI process as well.

Reply:
@charmik-redhat Good point about M1, I actually don't know anything about the state of things there. If we pretend that M1 problem does not exist, and we only talk about x86-64, here's a couple observations:
* `DEBUG_BUILD=yes make image` should work with both `deploy-local.sh` and `deploy.sh` without much difference.
* `deploy-local.sh` internally calls `deploy.sh`, the only difference is that `deploy-local.sh` disables monitoring to reduce the resource needs on the local cluster.
* You can use `deploy-local.sh` against remote clusters and `deploy.sh` against local clusters (as long as your local k8s has enough resources and deployments get scheduled successfully) without problems.
* You're right that you need to have images in quay repo when you deploy to a remote cluster. You are right, that `-debug` substring in the branch name enables debug build. You **should** also be able to push image built with `DEBUG_BUILD=yes make image` to quay.io/rhacs-eng/... . If you don't have permissions to do so, ask someone with permissions to add you to `eng` team there (https://quay.io/organization/rhacs-eng/teams/eng).